### DISCUSSION TITLE
Custom property returns empty using IPC socket

### DISCUSSION THREAD

**Discussion by KucharczykL** [mpv-player/mpv#14662] [Q&A] [answered]
Hello,

I am working on a custom script that sets the rating tag (via mutagen) for the currently playing file. I have that working and now I want to read the value using IPC socket. To do that, I thought using a custom property might work since I couldn't get `script-message` to return the data. However when I use the `read_property` command asking for `script-opts` I get empty `data` in the returned JSON.

```console
[rate_file] Set script-opts/rating to ['1']
```
```console
echo '{ "command": ["get_property", "script-opts"] }' | socat - /tmp/mpvsocket
{"data":{},"request_id":0,"error":"success"}
```

The only topic I found discussing getting custom properties via IPC is https://github.com/mpv-player/mpv/issues/10732 and the method there doesn't seem to work anymore. What am I doing wrong?

**Comment by guidocella** [comment, marked as answer]
`script-opts` doesn't support setting or accessing sub-properties with `/`. You can use `user-data/rating`.

**Reply by KucharczykL** [reply]
Ah, thanks so much, it works now!
